### PR TITLE
HIV-328: Request to change the edit/transfer section in program manag…

### PR DESCRIPTION
**Pull request by webtrendzs**
…er module

### CODE REVIEW

**Review by maikofelix47 — DISMISSED** [review]

File: src/app/etl-api/patient-program-resource.service.ts [hunk 1]
```diff
@@ -19,7 +21,14 @@ export class PatientProgramResourceService {
   public getPatientProgramVisitConfigs(patientUuid: string): Observable<any> {
     let url = this.appSettingsService.getEtlRestbaseurl().trim();
     url += 'patient-program-config?patientUuid=' + patientUuid;
-    return this.http.get(url);
+    return this.http.get(url).pipe(retryWhen((error) => {
+      return error.pipe(flatMap((_error: any) => {
```
Comment (enyachoke) [inline src/app/etl-api/patient-program-resource.service.ts]: Avoid using underscores. This is only allowed in get setters

File: src/app/patient-dashboard/common/formentry/patient-transfer.service.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    }
+    return this.transferState;
+  }
+
+  public prefillTransferOptions(): void {
+
+    const careStatus = this.searchNodeByQuestionId('careStatus');
+    if (careStatus.length > 0) {
+      // some delay just to ensure the question has been rendered then simulate a click before setting value.
+      setTimeout(() => {
```
Comment (enyachoke) [inline src/app/patient-dashboard/common/formentry/patient-transfer.service.ts:108]: Why did you need to use setTimeout here

File: src/app/patient-dashboard/common/formentry/patient-transfer.service.ts [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        const element: HTMLElement = document.getElementById('careStatusid') as HTMLElement;
```
Comment (enyachoke) [inline src/app/patient-dashboard/common/formentry/patient-transfer.service.ts:109]: Is it possible to use native angular methods instead of Vanillajs. 
Avoid dom manipulation inside a service.

File: src/app/patient-dashboard/common/formentry/patient-transfer.service.ts [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        element.click();
+        careStatus[0].control.setValue(localStorage.getItem('careStatus'));
+      }, 100);
+
+      // give some time for the hidden options to be shown
+      setTimeout(() => {
```
Comment (enyachoke) [inline src/app/patient-dashboard/common/formentry/patient-transfer.service.ts:118]: It it possible to avoid this setTimeout()?

File: src/app/patient-dashboard/general-landing-page/landing-page.component.ts [hunk 5]
```diff
@@ -286,6 +232,57 @@ export class GeneralLandingPageComponent implements OnInit, OnDestroy {
...
+          if (configs) {
+            this.allProgramVisitConfigs = configs;
+            _.each(this.enrolledProgrames, (program, index) => {
+              const referral = reply[index];
+              if (referral) {
+                _.extend(program, referral, {
+                  referral_completed: !_.isNil(referral.notification_status)
+                });
+                if (this.patientHasBeenSeenInProgram(program)) {
+                  console.log('Patient seen');
```
Comment (enyachoke) [inline src/app/patient-dashboard/general-landing-page/landing-page.component.ts]: You can remove this.

File: src/app/patient-dashboard/common/formentry/patient-transfer.service.ts (hunk 2 above)
Comment (webtrendzs) [reply src/app/patient-dashboard/common/formentry/patient-transfer.service.ts:108]: If you do it immediately, its not possible to set the value. But I can reduce the timeout delay. Also I did not find a better way other than to simulate a click event before setting the value. You know the funny delay when selecting options in form entry? Very evident on tablets.

File: src/app/patient-dashboard/common/formentry/patient-transfer.service.ts (hunk 3 above)
Comment (webtrendzs) [reply src/app/patient-dashboard/common/formentry/patient-transfer.service.ts:109]: I agree. I let this slip in intentionally when I hit a snag regarding the delay in getting the options by the form renderer. I am open for suggestions.

File: src/app/patient-dashboard/common/formentry/patient-transfer.service.ts (hunk 4 above)
Comment (webtrendzs) [reply src/app/patient-dashboard/common/formentry/patient-transfer.service.ts:118]: Please see my comments

File: src/app/patient-dashboard/general-landing-page/landing-page.component.ts (hunk 5 above)
Comment (webtrendzs) [reply src/app/patient-dashboard/general-landing-page/landing-page.component.ts]: please see my comments

File: src/app/etl-api/patient-program-resource.service.ts (hunk 1 above)
Comment (webtrendzs) [reply src/app/etl-api/patient-program-resource.service.ts]: done

**Review by kagai — APPROVED** [review]

**Review by maikofelix47 — APPROVED** [review]
